test/PEXv2.spec.ts

```diff
@@ -360,4 +361,61 @@ describe('evaluate', () => {
     expect(result).toEqual([{ message: 'ok', status: 'info', tag: 'root' }]);
     expect(pd.input_descriptors![0].constraints!.fields![0].filter!['_enum' as keyof FilterV2]).toEqual(['red']);
   });
+
+  it('should return ok if presentation definition @ in path works properly', () => {
+    const pd: PresentationDefinitionV2 = getPresentationDefinitionV2_3();
+    pd.input_descriptors[0].constraints!.fields = [
+      {
+        path: ['$.@context', '$.vc.@context'],
+        purpose:
+          'We can only verify bank accounts if they are attested by a trusted bank, auditor, or regulatory authority.',
+        filter: {
+          type: 'string',
+          _const: 'https://eu.com/claims/DriversLicense',
+        },
+      },
+    ];
+    const pex: PEXv2 = new PEXv2();
+    const vpSimple = getFile('./test/dif_pe_examples/vp/vp_general.json') as IVerifiablePresentation;
+    const result = pex.evaluateCredentials(pd, vpSimple.verifiableCredential, [], LIMIT_DISCLOSURE_SIGNATURE_SUITES);
+    console.log(JSON.stringify(result, null, 2));
+  });
+
+  it('should return ok if presentation definition @ in path escapes properly', () => {
+    const pd: PresentationDefinitionV2 = getPresentationDefinitionV2_3();
+    pd.input_descriptors[0].constraints!.fields = [
+      {
+        path: ["$..['@context']", "$.vc..['@context']"],
+        purpose:
+          'We can only verify bank accounts if they are attested by a trusted bank, auditor, or regulatory authority.',
+        filter: {
+          type: 'string',
+          _const: 'https://eu.com/claims/DriversLicense',
+        },
+      },
+    ];
+    const pex: PEXv2 = new PEXv2();
+    const vpSimple = getFile('./test/dif_pe_examples/vp/vp_general.json') as IVerifiablePresentation;
+    const result = pex.evaluateCredentials(pd, vpSimple.verifiableCredential, [], LIMIT_DISCLOSURE_SIGNATURE_SUITES);
+    console.log(JSON.stringify(result, null, 2));
+  });
+
+  it("other valid paths in json-ld shouldn't be affected by regex subs", () => {
+    const pd: PresentationDefinitionV2 = getPresentationDefinitionV2_3();
+    pd.input_descriptors[0].constraints!.fields = [
+      {
+        path: ['$..book[(@.length-1)]', '$..book[?(@.price<30 && @.category=="fiction")]', '$..book[?(@.price==8.95)]'],
+        purpose:
+          'We can only verify bank accounts if they are attested by a trusted bank, auditor, or regulatory authority.',
```

Comment:
Can we adjust the purpose. Because the path has no relationship with the purpose

Reply:
changed